src/FakeItEasy/Expressions/ParsedCallExpression.cs

```diff
@@ -34,5 +34,42 @@ internal class ParsedCallExpression
         public IEnumerable<ParsedArgumentExpression> ArgumentsExpressions { get; }
 
         public object CallTarget => this.callTarget.Value;
+
+        // Expression evaluation optimized for a very common case,
+        // locating the call target described by a local lambda closure
+        // such as
+        //   A.CallTo(() => fakeObj.doSomething(args))
+        //
+        // In this example, fakeObj is a field on the anonymous
+        // type that the compiler creates to represent the closure,
+        // so the expression will be a MemberExpression where the Member is
+        // a FieldInfo. That FieldInfo can be used to find the value of
+        // fakeObj, which is represented in the expression tree by the
+        // ConstantExpression fakeObj.

```

Comment:
Slight inaccuracy: "That FieldInfo can be used to find the value of  
 47   +        // fakeObj, which is represented in the expression tree by the  
 48   +        // ConstantExpression fakeObj.  
"

Slightly inaccurate with what it says about the constant expression: the ConstantExpression's value is equal to some anonymous object: the lambda closure.